controllers/nutanixcluster_controller.go

```diff
@@ -362,7 +375,7 @@ func (r *NutanixClusterReconciler) reconcileCredentialRefDelete(ctx context.Cont
 
 	if secret.DeletionTimestamp.IsZero() {
 		log.Info(fmt.Sprintf("removing secret %s in namespace %s for cluster %s", secret.Name, secret.Namespace, nutanixCluster.Name))
-		if err := r.Client.Delete(ctx, secret); err != nil && !errors.IsNotFound(err) {
+		if err := r.Client.Delete(ctx, secret); err != nil && !k8sApiErrors.IsNotFound(err) {
```

Comment:
we need to remove it from cache too here

Reply:
Good catch. Added a delete from cache towards the end of reconcileDelete.